### ISSUE TITLE
M-x eaf-install fails on gentoo; attempts to use apt

### ISSUE THREAD

**Issue by sio-k** [emacs-eaf/emacs-application-framework#924]
**Describe the bug**
`M-x eaf-install-and-update` attempts to use `apt` to install dependencies, fails when it can't find `apt`. This is an issue in `install-eaf.py/install_sys_deps()`, which can only cope with a very limited set of distros.

**To Reproduce**
Run `M-x eaf-install-and-update` on gentoo.

**Expected behavior**
`install-eaf.py/install_sys_deps()` should list dependencies and ask the user whether they are installed, then take the user at their word that the dependencies are, indeed, installed.

**Versions (please complete the following info):**
 - Distro and DE/WM: Gentoo with Kernel 5.12, XMonad 0.15-r1
 - Versions of Dependencies: don't matter; this is a problem related to installing those dependencies
 - M-x emacs-version: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.29, cauro version 1.16.0) of 2021-07-15

**Error logs**
`*eaf*` buffer:
```
Traceback (most recent call last):
  File "/home/.emacs.d/elpa/develop/eaf-20220214.36/eaf.py", line 25, in <module>
    from PyQt5 import QtWebEngineWidgets as NeverUsed # noqa
ModuleNotFoundError: No module named 'PyQt5'

Process *eaf* exited abnormally with code 1
```

**Comment by MatthewZMD** [comment]
> install-eaf.py/install_sys_deps() should list dependencies and ask the user whether they are installed, then take the user at their word that the dependencies are, indeed, installed.

There are many dependencies that it is infeasible to ask the user to know which deps they've installed already on the computer, running the install command for these dependencies should do no harm.

 The problem with your situation is, why do u have apt on gentoo? Are you able to use apt to install packages?

**Comment by sio-k** [comment]
No, I don't have apt installed. The problem is that the install script can't make heads nor tails of gentoo, and as such assumes it to be a debian derivative.

**Comment by sio-k** [comment]
The workaround I used to get EAF installed was to manually `emerge` the dependencies, then replace the contents of `install-eaf.py`'s `install_sys_deps()` with `pass`, which then caused the installation to succeed.

**Comment by MatthewZMD** [comment]
> No, I don't have apt installed. The problem is that the install script can't make heads nor tails of gentoo, and as such assumes it to be a debian derivative.

It will assume a debian/ubuntu derivative only when you have `apt` in your system by simply checking for `which apt`, and when it succeeds, it installs the dependencies using `apt install`.

Could you have a look at the `dependencies.json` and add the Gentoo dependencies using the same format with `emerge`, then create a PR?

**Comment by manateelazycat** [comment]
> The workaround I used to get EAF installed was to manually `emerge` the dependencies, then replace the contents of `install-eaf.py`'s `install_sys_deps()` with `pass`, which then caused the installation to succeed.

Below is EAF all depend for debian:

Core:

"apt": [
        "wmctrl",
        "nodejs",
        "npm",
        "libglib2.0-dev",
        "python3-pyqt5",
        "python3-sip",
        "python3-pyqt5.qtwebengine",
        "python3-pygit2"
    ]

File-Browser:
    "apt": [
        "python3-qrcode"
    ],

Markdown-Previewer:
    "apt": [
        "python3-markdown"
    ],

RSS-Reader:
    "apt": [
        "python3-pyquery"
    ],

File-Sender:
    "apt": [
        "python3-qrcode"
    ],

Music:
    "apt": [
        "libtag1-dev"
    ],

Airshare:
    "apt": [
        "python3-qrcode"
    ],

Jupyter:
    "apt": [
        "python3-qtconsole"
    ],

Browser:
  "apt": [
    "aria2"
  ],

File-Manager:
  "apt": [
    "fd-find"
  ],

Can you tell us which package in Gentoo? Then we can build a patch for Gentoo install.

**Comment by manateelazycat** [comment]
@sio-k I have add Gentoo support, can you try again? I just search package from https://packages.gentoo.org/ , not install Gentoo for test.

Any feedback are welcome. ;)

**Comment by sio-k** [comment]
Looks good. There's a few more deps you're missing:
- `net-libs/nodejs`
- `dev-python/PyQt5`
- `dev-python/PyQt5-sip` (though QtWebEngine might depend on both of those, you list them explicitly for debian)
- `dev-libs/glib`

I've also found out that an executable called `apt` is indeed installed on my system. It's definitely not Debian's APT, though.
Everything else looks good and `M-x eaf-install-and-update` runs through correctly now. Thank you.
